File: Leetcode/633.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long c_l(c);
        for (long long i = 0; i * i <= c_l; ++i) {
            double temp = sqrt(static_cast<double>(c_l - i * i));
            auto temp_l = static_cast<long long>(temp);
            if (temp_l * temp_l + i * i == c_l) {
                return true;
            }
        }
        return false;
    }
};

File: Leetcode/633.go
func judgeSquareSum(c int) bool {
	var c64 = int64(c)
	for i := int64(0); i * i <= c64; i++ {
		var j = int64(math.Sqrt(float64(c64 - i * i)))
		if j * j + i * i == c64 {
			return true
		}
	}
	return false
}